cpp/src/umap/runner.cuh

```diff
@@ -163,6 +163,167 @@ void _fit(const raft::handle_t& handle,
   ML::POP_RANGE();
 }
 
+template <typename value_idx, typename value_t, typename umap_inputs, int TPB_X, typename T>
+void _get_graph(const raft::handle_t& handle,
+                const umap_inputs& inputs,
+                UMAPParams* params,
+                T* cgraph_coo)
```

Comment:
Hey @cjnolet, slight issue using this, Since the template to `get_graph` passes `value_idx` as `ML::knn_indices_dense_t`. None of the `UMAPAlgo::` functions accept `raft::sparse::COO<float, ML::knn_indices_dense_t> *` as an input. 

However, using `COO<value_t>` as input (COO<value_t, int>) works fine. Would that be okay?

Reply:
Yeah, there's somewhat of an implicit conversion in the fuzzy simplicial set computation that demotes the double precision integer indexing to a single integer. We actually never needed the double precision but at the time it was being enforced by FAISS. That's since been updated on the FAISS side but we need to update it in the individual algorithms as well.

I have a PR going into RAFT that should make this update much easier to do across the algorithms but for now we can just assume single-precision int since that's already hardcoded in other places. It would also be useful to put a tiny comment somewhere around that argument that it assumes `int` for now, mostly for future eyes who might end up doing that cleanup.